server/strategies/authenticate.js

```diff
@@ -0,0 +1,66 @@
+import passport from 'passport'
+import { isAccountExpired } from '../utils/passport/helpers'
+import UserModel from '../models/UserModel'
+import ConfigModel from '../models/ConfigModel'
+
+export default async function (req, res, next) {
+  passport.authenticate(
+    'local-login',
+    { session: true },
+    async function (err, user) {
+      try {
+        if (err) return res.status(400).json({ error: err.message })
+
+        passport.serializeUser(function (user, done) {
+          done(null, user.uid)
+        })
+
+        passport.deserializeUser(function (user, done) {
+          done(null, user)
+        })
+
+        req.login(user, async function (err) {
+          if (err) return res.status(400).json({ error: err.message })
+
+          const { appDb, dataDb } = req.app.locals
+          const uid = user.uid
+          const configQuery = { owner: uid }
+          const config = await ConfigModel.findOne(appDb, configQuery)
+
+          if (!config) {
+            const configAttributes = { owner: uid, readers: [uid] }
+
+            await ConfigModel.create(appDb, configAttributes)
+          }
+
+          const userInfo = await UserModel.update(appDb, dataDb, uid, {
+            last_logon: Date.now(),
+          })
+          const { role, display_name, mail, icon, access, account_expires } =
+            userInfo
+
+          if (isAccountExpired(account_expires, role)) {
+            console.log('we should not be here')
+            await req.session.destroy()
+            await req.logout()
+
+            res.clearCookie('connect.sid')
+
+            return res.status(401).json({ error: 'Account is expired' })
+          }
+          req.session.role = role
+          req.session.display_name = display_name
+          req.session.mail = mail
+          req.session.celery_tasks = []
+          req.session.icon = icon
+          req.session.userAccess = access
```

Comment:
do we still need all this data stored on the session or can we just store the user identifying info so we can retrieve the user from the DB?

Reply:
[a17906c](https://github.com/AMP-SCZ/dpdash/pull/563/commits/a17906cb0c5882bfe9973da6840eb32502b4eef6)

Good point we no longer need it